modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java

```diff
@@ -518,7 +518,14 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
         if (!Boolean.TRUE.equals(model.isEnum)) {
             model.imports.add("JsonProperty");
             if (Boolean.TRUE.equals(model.hasEnums)) {
-                model.imports.add("JsonValue");
+                //Loop through properties to check if one is an embedded enum
+                for (CodegenProperty prop : model.allVars) {
+                    //If a property is an enum and a primitive type, it is an embedded enum
+                    if (prop.isEnum && prop.isPrimitiveType) {
+                        model.imports.add("JsonValue");
```

Comment:
This will have already occurred on line 519.

Reply:
Not what I witnessed: if the model itself is not an enum, but has variables that are enums, then kotlin generator need to have that import.

... Now that I reread, I see it's `hasEnums` so you should be right. Two possibilities: my code here changed nothing so I'll remove it, or `hasEnum`was `false` and I need to check out why. I'll try to do that on monday and report back

Reply:
@jimschubert Just took a look at it again, line 519 adds `JsonProperty` if a model is not an enum, because it will have properties.
My code replaces the previous line 521 that was adding `JsonValue` whenever a model has enums by a code that adds it only if one of the enum is embedded, as what was done previously by that generator. I did it to have the same result as master, although the `JsonValue` import doesn't seem to be used at all...